drivers/hmis/app/jobs/hmis/migrate_assessments_job.rb

```diff
@@ -0,0 +1,192 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class MigrateAssessmentsJob < ApplicationJob
+    include Hmis::Concerns::HmisArelHelper
+
+    EXIT_STAGE = 3
+    ENTRY_EXIT = [1, 3].freeze # entry, exit
+    NON_ENTRY_EXIT = [2, 5, 6].freeze # update, annual, post-exit
+    RELATED_RECORDS = [
+      Hmis::Hud::IncomeBenefit,
+      Hmis::Hud::HealthAndDv,
+      Hmis::Hud::EmploymentEducation,
+      Hmis::Hud::YouthEducationStatus,
+      Hmis::Hud::EnrollmentCoc,
+      Hmis::Hud::Disability,
+      Hmis::Hud::Exit,
+    ].freeze
+
+    # Construct CustomAssessment and FormProcessor records for Assessment-related records.
+    #
+    # For Entry/Exit assessments, records are grouped together if they have the same Data Collection Stage.
+    # If information dates differ across records, the earliest one is chosen.
+    #
+    # For Update/Annual/PostExit assessments, records are grouped together if they have the same
+    # Data Collection Stage AND information date.
+    #
+    # The resulting Assessments are constructed with:
+    #  - DateCreated = earliest creation date of related records
+    #  - DateUpdated = latest update date of related records
+    #  - UserID = randomly selected UserID from the collection of records
```

Comment:
Maybe we should grab the UserID from the latest DateUpdated?

Reply:
yeah, yeah it should. updating.